core/CMakeLists.txt

```diff
@@ -44,6 +44,32 @@ else()
         -Wall -Wextra -Wno-missing-braces -Werror=undef -Werror=return-type)
 endif()
 
+include (CheckCXXCompilerFlag)
+
+function(find_project_root result_variable)
+    set(current_root ${CMAKE_CURRENT_SOURCE_DIR})
+    while(NOT "${current_root}" STREQUAL "")
+        if(EXISTS "${current_root}/LICENSE")
+            break()
+        endif()
+        get_filename_component(current_root ${current_root} DIRECTORY)
+    endwhile()
+    set(${result_variable} ${current_root} PARENT_SCOPE)
+endfunction()
+
+find_project_root(PROJECT_ROOT)
```

Comment:
I think (and hope) that what you are looking for is simply `CMAKE_SOURCE_DIR`. This function does not sit well with me.

Reply:
I'll check, thanks for the hint.

Reply:
`CMAKE_SOURCE_DIR` is also `/home/ago/src/zxing-cpp/wrappers/python` when building the Python wrapper. 

I want to strip out the path prefix from the `__FILE__` log messages, but when building the Python wrapper some precompiled headers end up with `/home/ago/src/zxing-cpp` prefix due to GCC resolving the symlinks.

I'll think about alternative approaches.

Reply:
Why are you symlinking core in the wrapper instead of just calling `add_subdirectory` on the non-symlinked path?

I just tried and that seems to work great. I'll push an update with this.

Reply:
I think this is resolved.

Reply:
> Why are you symlinking core in the wrapper instead of just calling `add_subdirectory` on the non-symlinked path?

Because this is part of the fix that made sure the c++ source is part of the python source package. Hence your update will break this.

Which directory is your top-level cmake dir? Are you using setup.py to build your module?

Reply:
> Because this is part of the fix that made sure the c++ source is part of the python source package. Hence your update will break this.

Sorry, I missed that. I've restored the symlink and just make it resolve the path if it is a symlink.

> Which directory is your top-level cmake dir? Are you using setup.py to build your module?

I'm not sure, I am using pip to install the package using `pip install wrappers/python`